src/stdlib/builtin.verona

```diff
@@ -175,3 +175,106 @@ primitive Pointer[T] {
   builtin move(parent: mut, src: Pointer[T] & imm, dst: Pointer[T] & imm, size: U64 & imm);
 }
 
+/**
+ * Dynamically resizable array.
+ *
+ * Codegen and the VM have special support for this class. The trace function
+ * will traverse the first _size elements of the _data field. The class'
+ * destructor clears the data and frees the underlying memory allocation.
+ */
+class Array[T] {
+  _data: Pointer[T] & imm;
+  _size: U64 & imm;
+  _capacity: U64 & imm;
+
+  create(): Array[T] & iso
+  {
+    var this = new Array;
+    this._data = Pointer.allocate(0);

```

Comment:
This is going to perform an allocation will you are saying has zero capacity, so it will just get deallocated. Why not pick a larger size, so the work is atleast useful.

I think really we want `snmalloc` to say the size it allocates for the size you request.  Then capacity can be based on `snmalloc`s size classes.  I don't propose you solve that on this PR, as that requires changes that might affect `snmalloc`, but perhaps leave a TODO, or raise some issues.

Reply:
Do we need an allocation if `_capacity` is zero? As we only access the Pointer is `_capacity` and `_size` are greater than 0.  Makes correctness harder to justify?